api/envoy/config/core/v3/protocol.proto

```diff
@@ -350,6 +350,12 @@ message Http1ProtocolOptions {
   // (inferred if not present), host (from the host/:authority header) and path
   // (from first line or :path header).
   bool send_fully_qualified_url = 8;
+
+  // [#not-implemented-hide:] Hiding so that field can be removed after BalsaParser is rolled out.
+  // If set, force HTTP/1 parser: BalsaParser if true, http-parser if false.
+  // If unset, HTTP/1 parser is selected based on
+  // envoy.reloadable_features.http1_use_balsa_parser.
+  google.protobuf.BoolValue use_balsa_parser = 9;
```

Comment:
Hm, so this is a better place to put it but I had kinda wanted a wrapper proto 
message MigrationOptions {
BoolValue use_baslsa
BoolValue use_oghttp2
BoolValue use_uhv
}
so then we'd have one base MigrationOptions and one thing to removed.  I'm going to say leave this as-is and let @envoyproxy/api-shepherds and @adisuissa decide what makes more sense.

Reply:
This sounds like a reasonable option.  Note that AFAIK UHV is controlled by a compile-time flag, therefore it will be not possible to override behavior with a proto field.  So we could only group `use_balsa` and `use_oghttp2` together.

Reply:
The UHV transition document mentions the possibility of multiple UHV runtime flags added in the near future, one for each behavioral change.  If that will be the case, a third option would be to group those into one message, but leave `use_balsa` and `use_oghttp2` at the `Http1ProtocolOptions` level.

Reply:
I assume we want a config flag here as you don't want to test this globally? Seems as shame to add config that we ultimately will remove? I guess if we hide it it's OK. Can you also add the "work in progress" annotation to make it very clear that it can be removed later?

Reply:
I thought that uhv is going to be an [extension](https://github.com/envoyproxy/envoy/blob/b812ed619961c5289802ded5df1df27beb77fbb2/api/envoy/extensions/upstreams/http/v3/http_protocol_options.proto#L176).
The changes to balsa may be similar, as long as we assume that both balsa and http-parser could be used.

If the plan is to migrate and remove the old parser, and this migration needs to be at the protocol-options-level, then I agree that a "work-in-progress" tag just for balsa makes sense here.

Reply:
Thank you for the suggestion.  Adding `work_in_progress`.